encoding/protobuf_source.go

```diff
@@ -0,0 +1,79 @@
+package encoding
+
+import (
+	"fmt"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/protoparse"
+)
+
+// ProtoDescriptorSource is a source of protobuf descriptor information. It can be backed by a FileDescriptorSet
+// proto (like a file generated by protoc).
+// Adapted from https://github.com/fullstorydev/grpcurl
+type ProtoDescriptorSource interface {
+	// FindSymbol returns a descriptor for the given fully-qualified symbol name.
+	FindSymbol(fullyQualifiedName string) (desc.Descriptor, error)
+}
+
+// ProtoDescriptorSourceFromProtoFiles creates a DescriptorSource that is backed by the named files,
+// whose contents are Protocol Buffer source files. The given importPaths are used to locate
+// any imported files.
+func ProtoDescriptorSourceFromProtoFiles(importPaths []string, fileNames ...string) (ProtoDescriptorSource, error) {
+	p := protoparse.Parser{
+		ImportPaths:      importPaths,
+		InferImportPaths: len(importPaths) == 0,
+	}
+	fds, err := p.ParseFiles(fileNames...)
```

Comment:
General note: I haven't gone through the `protoreflect` package carefully, but as this is going from file to `FileDescriptor` without protoc, is this not dangerous? If `protoc` is not used to create the `FileDescriptors`, a lot of testing will be needed for edge cases.